storybookjs/storybook
# Core: Gracefully handle error when parsing preview.js file

Closes #23604 






## What I did

When parsing Storybook metadata, there's a chance the preview file breaks in the CSF AST parsing step.
That's because users might have unconventional code in their preview.js file which is not parseable:
```
// .storybook/preview.js
import { baseStorybookPreviewConfig } from './some-config'

export default baseStorybookPreviewConfig;
```

This PR handles the error in case it happens, and also improves the error message when it comes to AST parsing errors.

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

1. Create a sandbox
2. Edit the preview.js file to be:
```
// .storybook/preview.js
import { baseStorybookPreviewConfig } from './some-config'

export default baseStorybookPreviewConfig;
```
3. Build the Storybook. It should not break.

### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
